web_src/js/features/imagediff.js

```diff
@@ -88,29 +91,26 @@ export function initImageDiff() {
 
     for (const info of imageInfos) {
       if (info.$image.length > 0) {
-        $.ajax({
-          url: info.path,
-          success: (data, _, jqXHR) => {
-            info.$image.on('load', () => {
-              info.loaded = true;
-              setReadyIfLoaded();
-            }).on('error', () => {
-              info.loaded = true;
-              setReadyIfLoaded();
-              info.$boundsInfo.text('(image error)');
-            });
-            info.$image.attr('src', info.path);
+        info.$image.on('load', () => {
+          info.loaded = true;
+          setReadyIfLoaded();
+        }).on('error', () => {
+          info.loaded = true;
+          setReadyIfLoaded();
+          info.$boundsInfo.text('(image error)');
+        });
+        info.$image.attr('src', info.path);
 
-            if (jqXHR.getResponseHeader('Content-Type') === 'image/svg+xml') {
-              const bounds = getDefaultSvgBoundsIfUndefined(data, info.path);
-              if (bounds) {
-                info.$image.attr('width', bounds.width);
-                info.$image.attr('height', bounds.height);
-                hideElem(info.$boundsInfo);
-              }
-            }
+        if (parseUrl(info.path).pathname.toLowerCase().endsWith('.svg')) {
```

Comment:
what if image doesnot ends with `.svg` but still a `svg`

Reply:
I seems possible for this to occur because of the typesniffer module. Likely need to echo the detected content-type into HTML.

Reply:
Content-type was good alternative.

Reply:
This is fixed in https://github.com/go-gitea/gitea/pull/27267/commits/8039365c48961873ded8a37406a3450291c463f0 and https://github.com/go-gitea/gitea/pull/27267/commits/9ae1c4a2212d5b2bc87507ed513cc027eaacefe7.